docs/current/guides/snippets/use-services/expose-host-services-to-container/index.ts

```diff
@@ -0,0 +1,25 @@
+import { connect, Client } from "@dagger.io/dagger"
+
+connect(
+  async (client: Client) => {
+    // expose host service on port 3306
+    const hostSrv = client.host().service([{ frontend: 3306, backend: 3306 }])
+
+    // create MariaDB container
+    // with host service binding
+    // execute SQL query on host service
+    const out = await client
+      .container()
+      .from("mariadb:10.11.2")
+      .withServiceBinding("db", hostSrv)
+      .withExec([
+        "/bin/sh",
+        "-c",
+        "/usr/bin/mysql --user=root --password=secret --host=db -e 'SELECT * FROM mysql.user'",
```

Comment:
I'm getting the following output:
```shell
15: exec docker-entrypoint.sh /bin/sh -c /usr/bin/mysql --user=root --password=secret --host=db -e 'SELECT * FROM mysql.user'
15: [75.1s] ERROR 2013 (HY000): Lost connection to server at 'handshake: reading initial communication packet', system error: 11
15: exec docker-entrypoint.sh /bin/sh -c /usr/bin/mysql --user=root --password=secret --host=db -e 'SELECT * FROM mysql.user' ERROR: process "docker-entrypoint.sh /bin/sh -c /usr/bin/mysql --user=root --password=secret --host=db -e 'SELECT * FROM mysql.user'" did not complete successfully: exit code: 1

14: tunnel /upstream.3306.sock:3306/tcp
14: [0.48s] 2023/10/17 08:39:48 10.87.0.17:43152 > handling
14: [0.48s] 2023/10/17 08:39:48 10.87.0.17:43152 > dialed /tmp/.buildkit-ssh-sock2100707405/ssh_auth_sock
14: [75.5s] 2023/10/17 08:41:03 10.87.0.17:43152 > copy downstream->upstream error read tcp 10.87.0.18:3306->10.87.0.17:43152: use of closed network connection
```

Any idea @vito ?

Reply:
AFAIK this happens when the server is not running on the host. Can you try running the docker command in the guide to start a host MariaDB server first, and then re-try the script?